examples/subgraph/src/mutations/src/index.ts

```
@@ -29,36 +31,44 @@ async function sendTx(tx: any, msg: string, context: any) {
   }
 }
 
-function getGravityContract(context: any) {
+async function getGravityContract(context: any) {
   const { ethereum } = context.thegraph.config
   const { Gravity } = context.thegraph.dataSources
+  const [file] = await context.thegraph.config.ipfs.get(context.thegraph.dataSources.Gravity.abi)
+  const abi = file.content.toString('utf8')
 
   const contract = new ethers.Contract(
-    Gravity.address, Gravity.abi, ethereum
+    Gravity.address, abi, ethereum.getSigner()
   )
   contract.connect(ethereum)
   return contract
 }
 
 async function createGravatar(_root: any, {options}: any, context: any) {
   const { displayName, imageUrl } = options
-  const gravity = getGravityContract(context)
-  const tx = gravity.createGravatar(displayName, imageUrl)
+  const gravity = await getGravityContract(context)
+  // const tx = gravity.createGravatar(displayName, imageUrl)
 
-  await sendTx(tx, "Creating Gravatar", context)
-  return await queryUserGravatar(context)
+  await gravity.createGravatar(displayName, imageUrl);
+  // await sendTx(tx, "Creating Gravatar", context)
+  // return await queryUserGravatar(context)
+  return null;
```

null? Don't we want to return the new Gravatar entity? This should throw an error since the mutation resolver definition has a return type, no? Maybe we need to change the definition to be `createGravatar(...) Gravatar!` instead of `... ) Gravatar`